portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,40 +15,103 @@
 package com.google.sps.servlets;
 
 import com.google.appengine.api.datastore.DatastoreService;
+import java.util.*;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  ArrayList<String> greetings = new ArrayList<String>(); //Creating arraylist 
-  public DataServlet() {
-    greetings.add("hello");
-    greetings.add("whats up?");
-    greetings.add("nice to see you!");
-  }
+
     
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    String greetingJson = new Gson().toJson(greetings); //set to Json
-    response.getWriter().println(greetingJson); //send response
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int userChoice = getChoice(request);
+    System.out.println("user choice: ");
+    System.out.println(userChoice);
+    System.out.println("number of existing comments: ");
+    System.out.println(results.countEntities());
+
+
+
+    //goal of this is to add userChoice number of comments to list
+    //if there are fewer than userChoice comments, add all comments then stop adding 
+    List<Comment> UserComments = new ArrayList<>();
+    int commentNum = 0;
+    for (Entity entity : results.asIterable()) {
+        if(commentNum < userChoice || commentNum < results.countEntities()){//this line doesn't recognize 'size()' as a valid method 
```

Comment:
Unfortunately not a good idea to do this here.  As mentioned in the chat, `countEntities()` would query the database and doing this in a loop means you're querying the database multiple times.  As a rule of thumb, try to limit the number of times you query a database since each time you do will require the database server to perform some actions for you (plus the time required to send the request and wait for the response from the database).

An alternative approach would be to do something like:
```
int total_comments_count = results.countEntities();
for (...) {
  if (... || commentNum < total_comments_count) {
  }
}
```

Reply:
Also you maybe don't need to check for the total_comments_count at all, since the `results.asIterable()` for loop would auto exit if it runs out of comments to pull.  Meaning commentNum will never be greater than countEntities.